code/game/machinery/mind_machine.dm

```diff
@@ -126,19 +126,26 @@
 				bluespaceConduit += 1
 				playsound(src, 'sound/weapons/blaster.ogg', 10, 1)
 				to_chat(user, "<span class='notice'>The crystal is assimilated into the conduit!</span>")
+				qdel(A)
+				nanomanager.update_uis(src)
+				return
 			if(istype(A, /obj/item/bluespace_crystal/flawless))
 				bluespaceConduit += 300
 				playsound(src, 'sound/weapons/blaster-storm.ogg', 75, 1)
 				to_chat(user, "<span class='notice'>The bluespace conduit flashes violently before calming to a glow.</span>")
+				qdel(A)
+				nanomanager.update_uis(src)
+				return
 			else
 				bluespaceConduit += 3
 				playsound(src, 'sound/weapons/blaster.ogg', 25, 1)
 				to_chat(user, "<span class='notice'>The crystal is assimilated into the conduit!</span>")
-			qdel(A)
-			nanomanager.update_uis(src)
-			return
+				qdel(A)
```

Comment:
i don't get this change.

Reply:
The order I had it in it was giving 4 uses per artificial crystal. I'm not actually sure how that even works but it's something to do with it being both a bluespace crystal and a bluespace crystal/artificial, so it was stacking them since it wasn't being deleted immediately.